### DISCUSSION TITLE
About Window Width and Height Setting.

### DISCUSSION THREAD

**Discussion by wf-soft** [canton7/Stylet#246] [Ideas]
@canton7  Hi!
How can I set some properties of the view before the view is not null and rendered, such as the width and height of the window and other user control properties, otherwise it may not render as expected

**Comment by canton7** [comment]
I'm afraid I don't follow the question. Can you include an example? Thanks!

**Reply by wf-soft** [reply]
I have a window that contains some common settings, shortcut keys, styles, etc. I want to switch different screens in this window, but their width and height are different. If I set the width and height by binding, I can't open it correctly in the center of the parent screen in ShowWindow, and I should be uncertain about the width and height before the window is displayed

**Reply by wf-soft** [reply]
![71baaf3cc0caa8448c708efac08aa7fb](https://user-images.githubusercontent.com/62228539/116049434-d7ca2d80-a6a8-11eb-8ae4-59bef9396794.png)
![98fa38611c1f8c45949990325b28468a](https://user-images.githubusercontent.com/62228539/116049439-d993f100-a6a8-11eb-9a97-cc94f07bc620.png)
![d7ff1942f86a9f4280dd3020867382a0](https://user-images.githubusercontent.com/62228539/116049441-da2c8780-a6a8-11eb-92db-0fe03c8c0857.png)

**Reply by wf-soft** [reply]
Can I add a delegate after the view is created and before rendering, such as:
`
protected override void OnViewInitialize(View view);
`

**Reply by wf-soft** [reply]
![7c9df546b40c8944bbcc1a8d5ac47539](https://user-images.githubusercontent.com/62228539/116050989-b10cf680-a6aa-11eb-9c7e-68822cb9bb11.png)

I tried to use onviewloaded, but it didn't work as expected

**Reply by wf-soft** [reply]
@canton7 I wonder if there are other ways

**Comment by canton7** [comment]
I still don't entirely follow what you're doing, and why you're having problems setting the width and height using two-way (?) bindings.

There is a hook when the view is attached, but `Screen` implements it in a way which can't be overridden, [see here](https://github.com/canton7/Stylet/blob/5e97f9e1ea8c45e9d7033e6e7c7ea8ce41c3420a/Stylet/Screen.cs#L230). 

What you can do is:

```cs
var view = (Window)Helper.Ioc.Get<IViewManager>().CreateAndBindViewForModelIfNecessary(win);
view.Width = ...
view.Height = ...
Helper.Ioc.Get<IWindowManager>().ShowWindow(win);
```

**Reply by wf-soft** [reply]
@canton7  Thank you. This is a good solution to my problem. Is there any redundant code when I write like this
![3b0639f275da134e8beb39a5e1a2cf02](https://user-images.githubusercontent.com/62228539/116052425-30e79080-a6ac-11eb-8029-c1c04165bc52.png)
